File: DAA/Practica1/Hanoi.java
package practicas.practica1;

import java.util.Scanner;

public class Hanoi {
	
	static void hanoi(int n, int origen, int destino){
				int nuevoDestino;
		if(n == 1){
			if((origen == 1 || origen == 3) && destino != 2){
				nuevoDestino = 2;
				System.out.println("Mueve disco "+n+" desde la torre "+origen+" a torre "+nuevoDestino);
				System.out.println("Mueve disco "+n+" desde la torre "+nuevoDestino+" a torre "+destino);
			}
			else System.out.println("Mueve disco "+n+" desde la torre "+origen+" a torre "+destino);
		}
		else{
			if(origen == 1 || origen == 3){
				if(destino == 2) {
					if(origen == 1) nuevoDestino = 3;
					else nuevoDestino = 1;
					hanoi(n-1,origen,destino);	
					hanoi(n-1,origen,nuevoDestino);
					System.out.println("Mueve disco "+n+" desde la torre "+origen+" a torre "+destino);
					hanoi(n-1,nuevoDestino,destino);
				}
				else{
					nuevoDestino = 2;
					hanoi(n-1,origen,nuevoDestino);
					hanoi(n-1,nuevoDestino,destino);
					System.out.println("Mueve disco "+n+" desde la torre "+origen+" a torre "+nuevoDestino);
					hanoi(n-1,origen,nuevoDestino);
					hanoi(n-1,nuevoDestino,origen);
					System.out.println("Mueve disco "+n+" desde la torre "+nuevoDestino+" a torre "+destino);
					hanoi(n-1,origen,nuevoDestino);
					hanoi(n-1,nuevoDestino,destino);
				}
				
			}
			// origen = 2
			else{
				if(destino == 1) nuevoDestino = 3;
				else nuevoDestino = 1;
				hanoi(n-1,origen,nuevoDestino);
				System.out.println("Mueve disco "+n+" desde la torre "+origen+" a torre "+destino);
				hanoi(n-1,nuevoDestino,origen);
				hanoi(n-1,origen,destino);
			}
		}
	}
	
	public static void main(String args[]){
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		sc.close();
		if(n <= 9) hanoi(n,1,3);
	}
}

File: DAA/Practica1/hanoi.py
# -*- coding: utf-8 -*-

def hanoi():  
    n = int(input());
    if(n < 0 or n > 9):
        return;
        
    else:
        if(n>1):
            hanoi2(n-1,'1','2');
            hanoi2(n-1,'2','3');
        
        paso1 = 'Mueve disco '+ str(n) +' desde torre 1 a torre 2'
        print(paso1)
        
        if(n>1):
            hanoi2(n-1,'3','2');
            hanoi2(n-1,'2','1');
        
        paso2 = 'Mueve disco '+ str(n) +' desde torre 2 a torre 3'
        print(paso2)
        
        if(n>1):
            hanoi2(n-1,'1','2');
            hanoi2(n-1,'2','3');

def hanoi2(n,origen,destino): 
    
    if(origen == '1'):
        destino2 = '3'
    
    elif(origen == '3'):
        destino2 = '1'
    
    else:
        if(destino == '1'):
            destino2 = '3'
        else:
            destino2 = '1'
    
    if(n>1):
        if(origen == '2'):
            hanoi2(n-1,origen,destino2)
        else:
            hanoi2(n-1,origen,destino)
            hanoi2(n-1,destino,destino2)
        
    pasoN = 'Mueve disco '+ str(n) +' desde torre '+ origen+' a torre '+destino
    print(pasoN)
        
    if(n>1):
        if(origen == '2'):
            hanoi2(n-1,destino2,origen)
            hanoi2(n-1,origen,destino)
        else:
            hanoi2(n-1,destino2,destino)
            
def main():
    hanoi();
    
if __name__ == "__main__":
    main();